src/components/tests/delivery-form.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { createMount } from '@material-ui/core/test-utils';
+import { RobotDeliveryForm } from '../delivery-form';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import fakePlaces from '../../mock/data/places';
+import React from 'react';
+import { TDeliveryRequest } from '../commands-panel';
+
+const mount = createMount();
+
+const buildWrapper = (fleetName: string, onClick: TDeliveryRequest) => {
+  const wrapper = mount(<RobotDeliveryForm requestDelivery={onClick} fleetNames={[fleetName]} />);
+  return wrapper;
+};
+
+describe('Form validation', () => {
+  let isRequestButtonClicked = false;
+  const onClick = (
+    pickupPlaceName: string,
+    pickupDispenser: string,
+    dropOffPlaceName: string,
+    dropOffDispenser: string,
+    pickupBehaviour?: RomiCore.Behavior,
+    dropOffBehavior?: RomiCore.Behavior,
+  ) => {
+    isRequestButtonClicked = true;
+  };
+
+  beforeEach(() => {
+    isRequestButtonClicked = false;
+  });
+
+  test('Initial values', () => {
+    // Just places selected
+    const wrapper = buildWrapper('SuperFleet', onClick);
+    expect(
+      wrapper.findWhere(
+        x => x.name() === 'input' && x.props().value === fakePlaces()['SuperFleet'][0],
+      ),
+    ).toBeTruthy();
+
+    expect(
+      wrapper.findWhere(
+        x => x.name() === 'input' && x.props().value === fakePlaces()['SuperFleet'][1],
+      ),
+    ).toBeTruthy();
+
+    expect(
+      wrapper
+        .find(`#pickupDispenser`)
+        .find('input')
+        .props().value,
+    ).toBe('');
+
+    expect(
+      wrapper
+        .find(`#dropoffDispenser`)
+        .find('input')
+        .props().value,
+    ).toBe('');
+
+    wrapper.unmount();
+  });
+
+  test('Dispensers cannot be empty', () => {
+    const wrapper = buildWrapper('FleetB', onClick);
+    wrapper.find('form').simulate('submit');
+    expect(
+      wrapper
+        .find('#pickupDispenser-helper-text')
+        .first()
+        .find('p')
+        .props().children,
+    ).toBe('Cannot be empty');
```

Comment:
I think it may be too pedantic to check for the exact error string, it can cause problems if we ever have localization in the future, changing the error message would also cause it to fail. Is checking for the existence of error enough? Should we starting looking at a localization library like https://www.npmjs.com/package/i18next and check the error message matches a translation dictionary?

Reply:
Oh wow, I didn't know that, you're right about the localization problems. Ok, I changed it and I'm checking just the error existence for now. 
I thought the following when I wrote those tests, if an input could have more than one possible error, how do I ensure that is failing in the correct scenario? but now that you've mentioned I think that if this level of detail isn't required right now, especially if we'll add localization. 
I agree with you, we should start looking for a localization library and implement it before checking the error messages in the tests.
I'll create a new GH issue about localization.

Thanks for the feedback

Reply:
https://github.com/osrf/romi-dashboard/issues/73